src/main/java/com/metamx/common/io/smoosh/FileSmoosher.java

```diff
@@ -252,7 +263,7 @@ private Outer getNewCurrOut() throws FileNotFoundException
     final int fileNum = outFiles.size();
     File outFile = makeChunkFile(baseDir, fileNum);
     outFiles.add(outFile);
-    return new Outer(fileNum, new BufferedOutputStream(new FileOutputStream(outFile)), maxChunkSize);
+    return new Outer(fileNum, new FileOutputStream(outFile).getChannel(), maxChunkSize);
```

Comment:
@navis looking at the modern OpenJDK implementation, FOS additionally closes FileDescriptor. I'm not sure if Channel closes it too or not, but anyway it's right to close an "outer" object (ie fos, because channel is obtained from fos in this case), because 1) in different (maybe future) version of OpenJDK, or in a different JDK this may be different, and fos closes something non trivial in addition to what channel closes.